mitodl/ol-django
# Update build code; convert from rye to uv

### What are the relevant tickets?

mitodl/hq#5850

### Description (What does it do?)

- Updates ol-django to use uv for project management
- Fixes some issues with the build process
- Adds a test/template app (ostensibly to test the release CI workflow, but should be handy in case we need to add something new to this too)

### How can this be tested?

Read the README.md since this now uses `uv` and things have changed because of it. However, you should be able to do some basic stuff without issue:
- Run tests
- Build things
- Bump the version
- Update the changelog

As per the original issue, this doesn't really fix anything with regard to renovate. The runners have changed appropriately but none of that has been tested. That'll get covered in a separate PR.
